[WIP] Implement r_set_is_subset

TheNeikos opened the pull request:
This PR will implement `r_set_is_subset`

TheNeikos committed:
Implemented ht_is_subset

matthiasbeyer commented:
Besides the comment, this looks good to me.

matthiasbeyer commented on src/libreset/ht/ht_is_subset.c:
```diff
@@ -0,0 +1,41 @@
+        while (i--) {
+            int j = denom_buckets;
+            while(j--) {
+                if (!avl_is_subset(&ht_a->buckets[denom_buckets * j+i].avl,
```
Style: Missing space around `+`.

matthiasbeyer replied:
Also: `a * b + c` or `a * (b + c)` ? Make it clear and use parentheses, please.

TheNeikos committed:
Added ht_is_subset tests

TheNeikos committed:
Fixes switched variables in ht_subset

The variables were switched, which lead to the wrong avl trees to be tested.

Suggested-by: Matthias Beyer <mail@beyermatthias.de>
